src/Data/Nat/Divisibility/Core.agda

```diff
@@ -35,7 +35,27 @@ record _∣_ (m n : ℕ) : Set where
   constructor divides
   field quotient : ℕ
         equality : n ≡ quotient * m
-open _∣_ using (quotient) public
+
+  quotient≢0 : .{{NonZero n}} → NonZero quotient
+  quotient≢0 rewrite equality = m*n≢0⇒m≢0 quotient
+
+  n≡m*quotient : n ≡ m * quotient
+  n≡m*quotient rewrite equality = *-comm quotient m
```

Comment:
I think all these definitely belong in the `Basic properties` subsection rather the definition? Given that it would probably make more sense to define these outside of the record definition? Especially as we're exporting them publicly in the end?

Actually another question! Presumably these don't need to go in the `Core` file at all, but can go in `Data.Nat.Divisibility` instead?

Reply:
> I think all these definitely belong in the `Basic properties` subsection rather the definition? Given that it would probably make more sense to define these outside of the record definition? Especially as we're exporting them publicly in the end?

So... that's an interesting design point, for which I confess I made the opposite choice ('wide' `record` rather than 'narrow' definition plus top module-level properties) based on the experience of two things:
* the 'fat' interfaces to `Structures` and `Bundles` in the `Algebra` hierarchy: you want these things to be always-in-scope the moment you have an instance (here, of the relation);
* the discussion of 'symmetric instances for equalities' between myself and @JacquesCarette on issue #2115 (where there's a separate/underlying issue regarding the use of `Σ`-types in design: they are, almost by definition/design, a *non*-extensible `record`);
* the sense that being a 'property' (basic or otherwise) is somehow something which is (at least potentially, if not actually), *abstract* wrt the implementation/representation choices of the underlying type, whereas these 'properties' are somehow 'more' basic still, viz. they are bits of *structure* which should be *automatic* by virtue of the representation (so this might be something about wide or narrow APIs to ADTs, which we've discussed before... ;-))... with a 'logician''s hat on, I'd like to distinguish between 'admissible' (properties) vs. 'derivable' (structure), and this was one attempt to explore that space...

But maybe the 'narrow' version is preferable? I confess I still don't have a good feel for the spectrum of trade-offs here, and it's not something I have seen much discussed in the library, style-guide, etc. or the literature... 

> Actually another question! Presumably these don't need to go in the `Core` file at all, but can go in `Data.Nat.Divisibility` instead?

Well again, my perception/instinct was that that these were `Core` properties, again because of a (possibly-faulty!?) understanding that `Core` things don't exist merely to uncouple complex dependency cycles, but also to reflect the semantic admissible (abstract)/derivable (concrete) distinction above...?

Reply:
But I'll refrain from any refactoring of `Data.Nat.Divisibility` here...!
(So there's another separate design issue about this PR: keeping its footprint small... ;-))

Reply:
Ah your discussion has highlighted to me that I think this might be another instance of "implicit" knowledge that I have that's being missed here. There's a crucial difference between this type of record and the ones in algebra. 

**Namely**: (ideally) abstract records should be thin, non-abstract records can be fatter.

The algebra records are clearly designed to be non-abstract. Indeed if they were abstract, they would be pretty useless in general. However, the divisibility record, as you touch upon above, in an ideal world should be abstract! Therefore all properties should live externally to the record, as we shouldn't be encouraging the user to inspect the implementation of the record?

Does that make sense?

Reply:
Ah... will need to think, but I (start to) see what you might be saying...

Reply:
Oh, that's a really good point, and I agree: abstract records should be thin, non-abstract ones can be fatter.

If I try to put my thoughts into words: abstract records define new concepts where the outside world has no business knowing how everything is actually represented. Non-abstract ones are a convenient bundling of information, which indeed can for further convenience bundle up some 'conservative extensions' that are frequently used.

Reply:
So... now in the midst for refactoring, and I'm finding it is still (slightly) more efficient to 'cache' the flipped equality in the record... so that `open m∣n` saves some equational reasoning. So... suggest keeping it, but renaming it to `equalityᵒ`?

Reply:
See the recent commits...

Reply:
All new properties now in `Data.Nat.Divisibility`, so resolving this conversation. But very helpful discussion as regards 'tacit' knowledge, and criteria for 'narrow' vs. 'wide' APIs, esp. in respect of how 'abstract they may, or may not, be...!